llvm/lib/YkIR/YkIRWriter.cpp

```diff
@@ -83,12 +83,36 @@ template <class C, class E> size_t getIndex(C *Container, E *FindElement) {
   return Idx;
 }
 
-#define GENERIC_INST_SERIALISE(LLVM_INST, LLVM_INST_TYPE, YKIR_OPCODE)         \
-  if (isa<LLVM_INST_TYPE>(LLVM_INST)) {                                        \
-    serialiseInstGeneric(LLVM_INST, YKIR_OPCODE);                              \
-    return;                                                                    \
-  }
+// A <BBIdx, InstrIdx> pair that Uniquely identifies an Yk IR instruction within
+// a function.
+using InstrLoc = std::tuple<size_t, size_t>;
```

Comment:
It would probably be easier to have the instructions be a contiguous `Vec<Instr>` and then `InstrLoc` is a just `size_t`. We could then have blocks being `Vec<usize>` (or similar), which would indicate "instructions from x to y constitute block 1" and so on.

Reply:
That's going to be harder to think about, and harder to implement in the deserialiser (which would then need to be stateful). I think we should consider that an optimisation.

Reply:
Why would it need to be stateful?

Reply:
Currently, each block is decoded in isolation, so if we keep the current encoding, then the state for all prior blocks would need to be passed around. Deku allows this, but if that's what we want it feels like a hack.

If we really want this, then the correct thing to do is change the encoding so that all instructions from all of the blocks of a function are clumped together, then we wouldn't need that state, but it's a big change. I reckon it's a day's work to get right (mostly because I'd have to rewrite the test where we manually encode a module -- that's a total pain)

Reply:
We don't need to do it in this PR, but maybe keep it as an idea for the not-too-distant future, because it might make quite a few things easier.

Reply:
OK.

I think it's a good idea as long as we can abstract it such that we as devs don't have any cognitive burden to find block instructions. I don't see why we couldn't.